pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -710,6 +710,34 @@ func (gce *GCECloud) EnsureLoadBalancer(clusterName string, apiService *api.Serv
 		glog.Infof("Target pool %v for Service %v/%v doesn't exist", loadBalancerName, apiService.Namespace, apiService.Name)
 	}
 
+	// Ensure health checks are created for this target pool to pass to createTargetPool for health check links
+	// Alternately, if the annotation on the service was removed, we need to recreate the target pool without
+	// health checks. This needs to be prior to the forwarding rule deletion below otherwise it is not possible
+	// to delete just the target pool or http health checks later.
+	var hcToCreate *compute.HttpHealthCheck
+	hcExisting, _ := gce.GetHttpHealthCheck(loadBalancerName)
+	if path, healthCheckNodePort := apiservice.GetServiceHealthCheckPathPort(apiService); path != "" {
+		if hcExisting == nil {
```

Comment:
this is dangerous, you could've receive any error and you're concluding that the hc doesn't exist so you need to recreat the targetPool. Suggest checking for explicitly `isHTTPErrorCode(err, http.StatusNotFound)`